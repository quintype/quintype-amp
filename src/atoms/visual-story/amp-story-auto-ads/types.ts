import { Config } from "../../../types/config";
import { Story } from "../../../types/story";

export interface AutoAdsTypes {
  config: Config;
  story: Story;
}
