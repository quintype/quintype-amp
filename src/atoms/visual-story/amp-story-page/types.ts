import { Config } from "../../../types/config";
import { Story, Card } from "../../../types/story";

export interface AmpStoryPageTypes {
  config: Config;
  story: Story;
  id: string;
  card: Card;
  children: JSX.Element[] | JSX.Element | React.ReactChildren | React.ReactChild;
  "auto-advance-after"?: string;
}
