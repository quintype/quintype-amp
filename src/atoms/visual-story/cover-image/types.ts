import { Story } from "../../../types/story";
import { Config } from "../../../types/config";

export interface CoverImageProps {
  story: Story;
  config: Config;
}
