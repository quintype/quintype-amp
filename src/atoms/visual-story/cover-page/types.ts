import { Story } from "../../../types/story";
import { Config } from "../../../types/config";

export interface CoverPageProps {
  story: Story;
  config: Config;
}
