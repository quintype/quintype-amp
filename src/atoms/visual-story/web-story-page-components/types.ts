import { Card } from "../../../types/story";

export interface WebStoryPageComponentsTypes {
  card: Card;
}
