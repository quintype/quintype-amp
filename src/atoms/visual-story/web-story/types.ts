import { Story } from "../../../types/story";
import { Config } from "../../../types/config";

export interface WebStoryTypes {
  children: JSX.Element[] | JSX.Element | React.ReactChildren | React.ReactChild;
  story: Story;
  config: Config;
}
