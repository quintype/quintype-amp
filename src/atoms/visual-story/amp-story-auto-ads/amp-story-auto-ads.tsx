import React, { Fragment } from "react";
import { Helmet } from "react-helmet";
import { withStoryAndConfig } from "../../../context";
import { AutoAdsTypes } from "./types";
import { getTargetingInfo } from "../../dfp-ad/helpers";

export const AmpStoryAutoAdsBase = ({ story, config }: AutoAdsTypes) => {
  const dataSlot = config.opts?.featureConfig?.visualStories?.ads?.doubleclick?.dataSlot;
  if (!dataSlot) return null;
  const adConfig = {
    "ad-attributes": {
      type: "doubleclick",
      "data-slot": dataSlot,
      json: getTargetingInfo({ config, story })
    }
  };
  return (
    <Fragment>
      <Helmet>
        <script
          async={undefined}
          custom-element="amp-story-auto-ads"
          src="https://cdn.ampproject.org/v0/amp-story-auto-ads-0.1.js"
        />
      </Helmet>
      <amp-story-auto-ads>
        <script type="application/json" dangerouslySetInnerHTML={{ __html: JSON.stringify(adConfig) }} />
      </amp-story-auto-ads>
    </Fragment>
  );
};

export const AmpStoryAutoAds = withStoryAndConfig(AmpStoryAutoAdsBase);
