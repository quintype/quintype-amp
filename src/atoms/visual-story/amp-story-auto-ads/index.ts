import { AmpStoryAutoAds } from "./amp-story-auto-ads";
export { AmpStoryAutoAds };
