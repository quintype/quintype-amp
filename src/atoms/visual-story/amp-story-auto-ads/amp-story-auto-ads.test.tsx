import React from "react";
import { visualStory, config } from "../../../__fixtures__";
import { AmpStoryAutoAdsBase } from "./amp-story-auto-ads";
import { shallow } from "enzyme";
import { Config } from "../../../types/config";

describe("amp-story-auto-ads", () => {
  it("should match snapshot if visualStories config is object", () => {
    const configWithAd: Config = config;
    configWithAd.opts = {
      featureConfig: {
        visualStories: {
          ads: {
            doubleclick: {
              dataSlot: "foo"
            }
          }
        }
      }
    };
    expect(shallow(<AmpStoryAutoAdsBase story={visualStory} config={configWithAd} />)).toMatchSnapshot();
  });
  it("should match snapshot if visualStories config is array of object", () => {
    const configWithAd: Config = config;
    configWithAd.opts = {
      featureConfig: {
        visualStories: [
          {
            ads: {
              doubleclick: {
                dataSlot: "foo"
              }
            }
          }
        ]
      }
    };
    expect(shallow(<AmpStoryAutoAdsBase story={visualStory} config={configWithAd} />)).toMatchSnapshot();
  });
  it("should not render if ad config not passed in featureConfig", () => {
    const modifiedConfig: Config = config;
    modifiedConfig.opts = {};
    const wrapper = shallow(<AmpStoryAutoAdsBase story={visualStory} config={modifiedConfig} />);
    expect(wrapper.find("amp-story-auto-ads").length).toBe(0);
  });
  it("should render if ad config passed, taking correct value from featureConfig", () => {
    const configWithAd: Config = config;
    configWithAd.opts = {
      featureConfig: {
        visualStories: {
          ads: {
            doubleclick: {
              dataSlot: "foo"
            }
          }
        }
      }
    };
    const wrapper = shallow(<AmpStoryAutoAdsBase story={visualStory} config={configWithAd} />);
    expect(wrapper.find("script[type='application/json']").html()).toEqual(
      '<script type="application/json">{"ad-attributes":{"type":"doubleclick","data-slot":"foo","json":"{\\"targeting\\":{\\"environment\\":[\\"production\\"],\\"publisher-name\\":[\\"vikatan\\"],\\"sections\\":[\\"entertainment\\"]}}"}}</script>'
    );
  });
});
