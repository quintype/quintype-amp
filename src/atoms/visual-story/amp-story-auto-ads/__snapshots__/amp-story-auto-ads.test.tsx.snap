// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amp-story-auto-ads should match snapshot if visualStories config is array of object 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <script
      custom-element="amp-story-auto-ads"
      src="https://cdn.ampproject.org/v0/amp-story-auto-ads-0.1.js"
    />
  </HelmetWrapper>
  <amp-story-auto-ads>
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{\\"ad-attributes\\":{\\"type\\":\\"doubleclick\\",\\"data-slot\\":\\"foo\\",\\"json\\":\\"{\\\\\\"targeting\\\\\\":{\\\\\\"environment\\\\\\":[\\\\\\"production\\\\\\"],\\\\\\"publisher-name\\\\\\":[\\\\\\"vikatan\\\\\\"],\\\\\\"sections\\\\\\":[\\\\\\"entertainment\\\\\\"]}}\\"}}",
        }
      }
      type="application/json"
    />
  </amp-story-auto-ads>
</Fragment>
`;

exports[`amp-story-auto-ads should match snapshot if visualStories config is object 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <script
      custom-element="amp-story-auto-ads"
      src="https://cdn.ampproject.org/v0/amp-story-auto-ads-0.1.js"
    />
  </HelmetWrapper>
  <amp-story-auto-ads>
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{\\"ad-attributes\\":{\\"type\\":\\"doubleclick\\",\\"data-slot\\":\\"foo\\",\\"json\\":\\"{\\\\\\"targeting\\\\\\":{\\\\\\"environment\\\\\\":[\\\\\\"production\\\\\\"],\\\\\\"publisher-name\\\\\\":[\\\\\\"vikatan\\\\\\"],\\\\\\"sections\\\\\\":[\\\\\\"entertainment\\\\\\"]}}\\"}}",
        }
      }
      type="application/json"
    />
  </amp-story-auto-ads>
</Fragment>
`;
